packages/react-devtools-shared/src/backend/views/Highlighter/index.js

```diff
@@ -151,14 +151,23 @@ export default function setupHighlighter(
     event.preventDefault();
     event.stopPropagation();
 
-    selectFiberForNode(((event.target: any): HTMLElement));
+    selectFiberForNode(
+      (((event.composed
+        ? event.composedPath()[0]
+        : event.target): any): HTMLElement),
+    );
   }
 
-  function onPointerOver(event: MouseEvent) {
+  let lastHoveredNode: HTMLElement | null = null;
+  function onPointerMove(event: MouseEvent) {
     event.preventDefault();
     event.stopPropagation();
 
-    const target = ((event.target: any): HTMLElement);
+    const target: HTMLElement = ((event.composed
+      ? event.composedPath()[0]
+      : event.target): any);
+    if (lastHoveredNode === target) return;
+    lastHoveredNode = target;
```

Comment:
If we could listen to `pointerover` event, do we need this kind of optimization? Or what is the purpose of it?

Reply:
this basically check if the target node  is the same as the lasthoverednode and if it is , then we are doing an early return for the same, this can help us to prevent extra dom calculations happening on it if the hovered node has not changed!